aiida_quantumespresso/calculations/projwfc.py

```diff
@@ -13,32 +10,42 @@ class ProjwfcCalculation(NamelistsCalculation):
     <Psi(n,k) | Y(theta,phi)R(r) >.
     For more information, refer to http://www.quantum-espresso.org/
     """
-    def _init_internal_params(self):
-        super(ProjwfcCalculation, self)._init_internal_params()
-        # self._PROJWFC_FILENAME = 'aiida.pdos'
-        self._default_namelists = ['PROJWFC']
-        self._blocked_keywords = [
-                                  ('PROJWFC','outdir',self._OUTPUT_SUBFOLDER),
-                                  ('PROJWFC','prefix',self._PREFIX),
-                                  ('PROJWFC','lsym',True),
-                                  ('PROJWFC','lwrite_overlaps',False),
-                                  ('PROJWFC','lbinary_data',False),
-                                  ('PROJWFC','kresolveddos',False),
-                                  ('PROJWFC','tdosinboxes',False),
-                                  ('PROJWFC','plotboxes',False),
-
-                                 ]
-        self._default_parser = 'quantumespresso.projwfc'
-        self._internal_retrieve_list = [self._PREFIX+".pdos*"]
-
-    def use_parent_calculation(self, calc):
-        """Set the parent calculation."""
-        if not isinstance(calc, PwCalculation):
-            raise ValueError('Parent calculation must be a PwCalculation')
-
-        try:
-            remote_folder = calc.get_outgoing(node_class=RemoteData, link_label_filter='remote_folder').one().node
-        except ValueError:
-            raise exceptions.UniquenessError('Parent calculation does not have a remote folder output node.')
-
-        self.use_parent_folder(remote_folder)
+    #_PROJWFC_FILENAME = 'aiida.pdos'
+    _default_namelists = ['PROJWFC']
+    _blocked_keywords = [
+        ('PROJWFC', 'outdir', NamelistsCalculation._OUTPUT_SUBFOLDER),
+        ('PROJWFC', 'prefix', NamelistsCalculation._PREFIX),
+        ('PROJWFC', 'lsym', True),
+        ('PROJWFC', 'lwrite_overlaps', False),
+        ('PROJWFC', 'lbinary_data', False),
+        ('PROJWFC', 'kresolveddos', False),
+        ('PROJWFC', 'tdosinboxes', False),
+        ('PROJWFC', 'plotboxes', False),
+    ]
+    _default_parser = 'quantumespresso.projwfc'
+    _internal_retrieve_list = [NamelistsCalculation._PREFIX + ".pdos*"]
+    
+    @classmethod
+    def define(cls, spec):
+        from aiida.orm import ProjectionData, BandsData
+        super(ProjwfcCalculation, cls).define(spec)
+        spec.input('parent_folder', valid_type=(RemoteData, FolderData), help='The output folder of a pw.x calculation')
+        spec.output('output_parameters', valid_type=Dict)
+        spec.output('Dos', valid_type=XyData)
```

Comment:
I saw that apparently the old implementation already used `Dos` with capital as well, but I think it is weird. Let's see if we can add this to the list of changes we want to make

Reply:
Agreed.
Thanks for the review.